java/code/src/com/suse/manager/attestation/test/AttestationManagerTest.java

```diff
@@ -136,34 +137,104 @@ public void testCreateAttestationAction() throws TaskomaticApiException {
     }
 
     @Test
-    public void testListReports() {
+    public void countAttestationReportsForUserAndSystem() {
         mgr.createConfig(user, server, CoCoEnvironmentType.KVM_AMD_EPYC_GENOA, true);
+        mgr.createConfig(user, server3, CoCoEnvironmentType.KVM_AMD_EPYC_GENOA, true);
 
-        ServerCoCoAttestationReport report = mgr.initializeReport(user, server);
-        mgr.initializeResults(user, report);
-        fakeSuccessfullAttestation(report);
+        createFakeAttestationReport(user, server);
+        createFakeAttestationReport(user, server);
+        createFakeAttestationReport(user, server3);
 
-        ServerCoCoAttestationReport report2 = mgr.initializeReport(user, server);
-        mgr.initializeResults(user, report2);
-        fakeSuccessfullAttestation(report2);
         HibernateFactory.getSession().flush();
         HibernateFactory.commitTransaction();
         HibernateFactory.getSession().clear();
+        commitHappened();
```

Comment:
Have you checked, that this really cleanup all the config, reports and results?

Reply:
Yes, toggling the `committed` flag will trigger the removal of the organization at the end of the process in [JMockBaseTestCaseWithUser](https://github.com/uyuni-project/uyuni/blob/98f5f0790457610509e637c8e5638e5e039063f8/java/code/src/com/redhat/rhn/testing/JMockBaseTestCaseWithUser.java#L63C1-L68C10):

```java
        // If at some point we created a user and committed the transaction, we need
        // clean up our mess
        if (committed) {
            OrgFactory.deleteOrg(user.getOrg().getId(), user);
            commitAndCloseSession();
        }
```